packages/api/src/sequelize/migrations/2025-08-13_00_alter_tcm-encounter_add_has-cardiac-codes.ts

```diff
@@ -0,0 +1,41 @@
+import type { Migration } from "..";
+import { cardiacCodes } from "../../command/medical/tcm-encounter/cardiac-codes";
+
+const tableName = "tcm_encounter";
+const columnName = "has_cardiac_code";
+
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async transaction => {
+    // Add generated column that automatically computes cardiac code presence
+    await queryInterface.sequelize.query(
+      `
+      ALTER TABLE ${tableName} 
+      ADD COLUMN ${columnName} BOOLEAN 
+      GENERATED ALWAYS AS (
+        jsonb_path_query_array(clinical_information, '$.condition[*].coding[*].code')
+        ?| ARRAY['${cardiacCodes.join("', '")}']
+      ) STORED;
+      `,
+      { transaction }
+    );
+  });
+
+  // Create index outside transaction (CONCURRENTLY cannot run inside transaction)
+  await queryInterface.sequelize.query(`
+    CREATE INDEX CONCURRENTLY idx_tcm_encounter_cardiac_code 
+      ON ${tableName} (${columnName}) 
+      WHERE ${columnName} = true;
+  `);
```

Comment:
Do we expect this to take long to process? We want to avoid triggering more than one at the same time... ECS gives the first new task some time and then runs the next one... if the first one didn't finish migrating, we'll run this again

Reply:
TIL: `CONCURRENTLY`